tools/Voron.Recovery/RecoveredDatabaseCreator.cs

```diff
@@ -0,0 +1,433 @@
+
+        private readonly DatabaseDestination _databaseDestination;
+        private readonly SmugglerResult _results;
+        private readonly DocumentsOperationContext _context;
+        private readonly DocumentDatabase _database;
+        private readonly IDisposable _contextDisposal;
+        private readonly Logger _logger;
+        private readonly string _logDocId;
+        private readonly string _orphanAttachmentsDocIdPrefix;
+        private readonly string _orphanCountersDocIdPrefix;
+        private readonly Slice _attachmentsSlice;
+        private readonly ByteStringContext _byteStringContext;
+        private readonly string _recoveryLogCollection;
+        private string GetOrphanAttachmentDocId(string hash) => $"{_orphanAttachmentsDocIdPrefix}/{hash}";
+
+        private RecoveredDatabaseCreator(DocumentDatabase documentDatabase, string recoverySession, Logger logger)
+        {
+            _logger = logger;
+            _recoveryLogCollection = $"RecoveryLog-{recoverySession}";
+            _orphanAttachmentsDocIdPrefix = $"OrphanAttachments/{recoverySession}";
+            _orphanCountersDocIdPrefix = $"OrphanCounters/{recoverySession}";
+            _logDocId = $"Log/{recoverySession}";
+            _database = documentDatabase;
+            _databaseDestination = new DatabaseDestination(documentDatabase);
+            _results = new SmugglerResult();
+            _databaseDestination.Initialize(new DatabaseSmugglerOptionsServerSide(), _results, ServerVersion.Build);
+            _contextDisposal = _database.DocumentsStorage.ContextPool.AllocateOperationContext(out _context);
+            _byteStringContext = new ByteStringContext(SharedMultipleUseFlag.None);
+            Slice.From(_byteStringContext, "Attachments", ByteStringType.Immutable, out _attachmentsSlice);
+
+            using (var tx = _context.OpenWriteTransaction())
+            {
+                using (var doc = _context.ReadObject(
+                    new DynamicJsonValue
+                    {
+                        ["RecoverySession"] = recoverySession,
+                        ["RecoveryStarted"] = DateTime.Now,
+                        [Constants.Documents.Metadata.Key] = new DynamicJsonValue {[Constants.Documents.Metadata.Collection] = _recoveryLogCollection}
+                    }, _logDocId, BlittableJsonDocumentBuilder.UsageMode.ToDisk)
+                )
+                {
+                    _database.DocumentsStorage.Put(_context, _logDocId, null, doc);
+                }
+                tx.Commit();
+            }
+        }
+
+        public void WriteDocument(Document document)
+        {
+            var actions = _databaseDestination.Documents();
+            WriteDocumentInternal(document, actions);
+        }
+
+        public void WriteRevision(Document document)
+        {
+            if (_database.DocumentsStorage.RevisionsStorage.Configuration == null)
```

Comment:
why are you dealing with the configuration in this manner?
If there are revisions in the DB, we want to preserve them.

It does bring a good point. Recovery won't help us with overall config (etl, revisions) or cluster wide items such as identity or compare exchange values.
Need to be sure that we make a note of this

Reply:
The recovered database is a new db. I am creating new config. Is there another way of doing that?

Reply:
This is again about not running a server. You can manually set this value to the right parameters, in memory only.

Reply:
Different idea. Perform a recovery on a live server and take database down? Copy values from Server Storage (Cluster), etc?

Reply:
Ping